File: MetroTips #18 Webカメラで撮影するには？/[C#][VB]-MetroTips #18 Webカメラで撮影するには？/C# and VB/C#,VB/MetroTips018VB/Common/BooleanToVisibilityConverter.vb
Namespace Common

    ''' <summary>
    ''' true を <see cref="Visibility.Visible"/> に、および false を 
    ''' <see cref="Visibility.Collapsed"/> に変換する値コンバーター。
    ''' </summary>
    Public NotInheritable Class BooleanToVisibilityConverter
        Implements IValueConverter

        Public Function Convert(value As Object, targetType As Type, parameter As Object,
                                language As String) As Object Implements IValueConverter.Convert
            If TypeOf value Is Boolean AndAlso DirectCast(value, Boolean) Then Return Visibility.Visible
            Return Visibility.Collapsed
        End Function

        Public Function ConvertBack(value As Object, targetType As Type, parameter As Object,
                                    language As String) As Object Implements IValueConverter.ConvertBack
            Return TypeOf value Is Visibility AndAlso DirectCast(value, Visibility) = Visibility.Visible
        End Function

    End Class

End Namespace

File: Reversi board game app in XAML, C#, and C++/[C#][C++]-Reversi board game app in XAML, C#, and C++/C# and C++/Reversi/Common/Converters/BooleanToVisibilityConverter.cs
using System;
using Windows.UI.Xaml;
using Windows.UI.Xaml.Data;

namespace Reversi.Common.Converters
{
    // NOTE: This class is slightly modified from the template-generated original
    // to add the reversability mechanism. 

    /// <summary>
    /// Value converter that translates true to <see cref="Visibility.Visible"/> and false to
    /// <see cref="Visibility.Collapsed"/>, or the reverse if the parameter is "Reverse".
    /// </summary>
    public sealed class BooleanToVisibilityConverter : IValueConverter
    {
        public object Convert(object value, Type targetType, object parameter, string language)
        {
            return ((value is bool && (bool)value) ^
                (parameter as string ?? string.Empty).Equals("Reverse")) ? 
                Visibility.Visible : Visibility.Collapsed;
        }

        public object ConvertBack(object value, Type targetType, object parameter, string language)
        {
            return (value is Visibility && (Visibility)value == Visibility.Visible) ^ 
                (parameter as string ?? string.Empty).Equals("Reverse");
        }
    }
}
